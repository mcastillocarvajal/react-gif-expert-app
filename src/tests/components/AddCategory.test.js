import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { AddCategory } from '../../components/AddCategory'


describe('Test to <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory 
        setCategories={ setCategories }
    />);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory 
            setCategories={ setCategories }
        />)
    })
    
    test('should show component correctly', () => {     
        expect( wrapper ).toMatchSnapshot();
    })

    test('should change the input box text', () => {
        
        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        input.simulate('change', { target: { value }});
    })

    test('should NOT post info in submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect( setCategories ).not.toHaveBeenCalled();
    })
    
    test('should call setCategories and clean input box', () => {
        
        const value = 'Hola Mundo';
        const input = wrapper.find('input')
        wrapper.find('input').simulate('change', { target: { value } })
        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
        expect( input.prop('value') ).toBe('');
    })
})
