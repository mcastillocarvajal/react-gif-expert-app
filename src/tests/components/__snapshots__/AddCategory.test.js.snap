// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to <AddCategory /> should show component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
